composables/transaction/transactionBurn.ts

```diff
@@ -14,11 +14,42 @@ export function execBurnTx(item: ActionConsume, api, executeTransaction) {
   }
 
   if (item.urlPrefix === 'snek' || item.urlPrefix === 'bsx') {
-    executeTransaction({
-      cb: getApiCall(api, item.urlPrefix, Interaction.CONSUME),
-      arg: bsxParamResolver(item.nftId, Interaction.CONSUME, ''),
-      successMessage: item.successMessage,
-      errorMessage: item.errorMessage,
+    const [collectionId, tokenId] = bsxParamResolver(
+      item.nftId,
+      Interaction.CONSUME,
+      ''
+    )
+    const hasOffers = ref(false)
+    const { data } = useGraphql({
+      queryName: 'offerListByNftId',
+      queryPrefix: 'chain-bsx',
+      variables: {
+        id: item.nftId,
+      },
+    })
+
+    watch(data, (data) => {
+      hasOffers.value = Boolean(data.offers.length)
+      const offerWithdrawArgs = data.offers.map((offer: { caller: string }) => {
+        return api.tx.marketplace.withdrawOffer(
+          collectionId,
+          tokenId,
+          offer.caller
+        )
+      })
+      const cb = hasOffers.value
+        ? api.tx.utility.batchAll
+        : getApiCall(api, item.urlPrefix, Interaction.CONSUME)
+      const arg = [
+        [...offerWithdrawArgs, api.tx.nft.burn(collectionId, tokenId)],
+      ]
+
+      executeTransaction({
+        cb: cb,
+        arg: arg,
+        successMessage: item.successMessage,
+        errorMessage: item.errorMessage,
+      })
```

Comment:
Missing `arg: bsxParamResolver(item.nftId, Interaction.CONSUME, ''),`

Reply:
> Missing `arg: bsxParamResolver(item.nftId, Interaction.CONSUME, ''),`

same thing as `api.tx.nft.burn(collectionId, tokenId)`

Reply:
Not the same sir